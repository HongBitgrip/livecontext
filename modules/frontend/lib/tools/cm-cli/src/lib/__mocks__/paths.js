'use srict';

const paths = jest.genMockFromModule('../paths');

paths.__setBricksDirectory = value => {
  paths.BRICKS_DIRECTORY = value;
};

paths.__setJSLibsDirectory = value => {
  paths.JSLIBS_DIRECTORY = value;
};

paths.__setToolsDirectory = value => {
  paths.TOOLS_DIRECTORY = value;
};

paths.__setThemesDirectory = value => {
  paths.THEMES_DIRECTORY = value;
};

module.exports = paths;
