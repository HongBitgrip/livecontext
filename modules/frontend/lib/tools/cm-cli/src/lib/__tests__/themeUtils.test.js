"use strict";

jest.mock("fs");
jest.mock("mkdirp");
jest.mock("../paths");

const log = {
  debug: jest.fn(),
  info: jest.fn(),
  warn: jest.fn(),
  error: jest.fn(),
  success: jest.fn(),
};

const wsConfig = {
  themesPath: "/path/to/themes/",
};

describe("convertThemeName()", () => {
  const { convertThemeName } = require("../themeUtils");
  it("returns converted theme name", () => {
    const themeName =
      ' 1234567890_-Ugly Theme Name! äöüß^°!"§$%&/()=?´`*+#.:,;<>@„¡“¶¢[]|{}”≠¿±‘’æœ•–…∞µ~√ç≈¥≤åº∆π¨Ω†®€∑«»';
    const expected = "1234567890-uglythemename";
    expect(convertThemeName(themeName)).toEqual(expected);
  });
  it("returns empty string", () => {
    const themeName =
      ' äöüß^°!"§$%&/()=?´`*+#.:,;<>@„¡“¶¢[]|{}”≠¿±‘’æœ•–…∞µ~√ç≈¥≤åº∆π¨Ω†®€∑«»';
    expect(convertThemeName(themeName)).toEqual("");
  });
  it("returns empty string", () => {
    const themeName = [1, 2];
    expect(convertThemeName(themeName)).toEqual("");
  });
  it("returns empty string", () => {
    const themeName = {};
    expect(convertThemeName(themeName)).toEqual("");
  });
});

describe("isThemeNameInUse()", () => {
  beforeEach(() => {
    jest.resetModules();
    require("fs").__setMockFiles([
      "/path/to/themes/aurora-theme",
      "/path/to/themes/corporate-theme",
      "/path/to/themes/hybris-theme",
    ]);
    require("../paths").__setThemesDirectory("/path/to/themes");
  });

  it("returns true", () => {
    const { isThemeNameInUse } = require("../themeUtils");
    expect(isThemeNameInUse("/path/to/themes/aurora-theme")).toBe(true);
  });
  it("returns false", () => {
    const { isThemeNameInUse } = require("../themeUtils");
    expect(isThemeNameInUse("/path/to/themes/newest-theme")).toBe(false);
  });
});

describe("createFolderStructure()", () => {
  it("creates folder structure of a new theme", () => {
    const mkdirp = require("mkdirp");
    mkdirp.__resetMockDirectories();

    const { createFolderStructure } = require("../themeUtils");
    createFolderStructure(wsConfig, "/path/to/themes/newest-theme", log);
    const directories = mkdirp.__getMockDirectories();
    expect(directories).toMatchSnapshot();
  });
});

describe("createFiles()", () => {
  it("creates files of a new theme", () => {
    const fs = require("fs");
    fs.__resetMockFiles();

    const { createFiles } = require("../themeUtils");
    createFiles("/path/to/themes/newest-theme", "newest", log);
    const files = fs.__getMockFiles();
    expect(files).toMatchSnapshot();
  });
});

describe("createTheme()", () => {
  it("creates folder structure and files of a new theme", () => {
    const mkdirp = require("mkdirp");
    mkdirp.__resetMockDirectories();
    const fs = require("fs");
    fs.__resetMockFiles();

    const { createTheme } = require("../themeUtils");
    createTheme(wsConfig, "/path/to/themes/newest-theme", "newest", log);

    const directories = mkdirp.__getMockDirectories();
    const files = fs.__getMockFiles();

    expect(directories).toMatchSnapshot();
    expect(files).toMatchSnapshot();
  });
});
