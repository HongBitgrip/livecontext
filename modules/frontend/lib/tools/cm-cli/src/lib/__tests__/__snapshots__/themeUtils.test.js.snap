// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`createFiles() creates files of a new theme 1`] = `
Object {
  "/path/to/themes/newest-theme": Object {
    "newest-theme.xml": "<?xml version=\\"1.0\\" encoding=\\"UTF-8\\"?>
<themeDefinition modelVersion=\\"1\\">
<name>newest</name>
<!-- <description>a short description</description> -->
<!-- <thumbnail>img/theme-preview.jpg</thumbnail> -->
<javaScriptLibraries>
  <!-- add thirdparty js files that are not bundled via webpack here -->
  <!-- <javaScript>vendor/some-vendor-script.js</javaScript> -->
</javaScriptLibraries>
<javaScripts>
  <!-- add own js files that are not bundled via webpack here -->
  <!-- <javaScript>js/your-code.js</javaScript> -->
  <javaScript disableCompress=\\"true\\">js/newest.js</javaScript>
</javaScripts>
<styleSheets>
  <css disableCompress=\\"true\\">css/newest.css</css>
</styleSheets>
<templateSets>
  <templateSet>templates/newest-templates.jar</templateSet>
  <templateSet>templates/bricks-templates.jar</templateSet>
</templateSets>
<resourceBundles>
  <!-- add theme resource bundles here -->
  <resourceBundle>l10n/Newest_en.properties</resourceBundle>
  <!-- merged resource bundles of all bricks, activate this if you are using bricks -->
  <!-- <resourceBundle>l10n/Bricks_en.properties</resourceBundle> -->
</resourceBundles>
</themeDefinition>
",
    "package.json": "{
  \\"name\\": \\"@coremedia/newest-theme\\",
  \\"version\\": \\"1.0.0\\",
  \\"private\\": true,
  \\"scripts\\": {
    \\"build\\": \\"webpack\\",
    \\"start\\": \\"cm monitor\\",
    \\"theme-importer\\": \\"cm theme-importer\\"
  },
  \\"__comment__dependencies__\\": {
    \\"__comment__\\": \\"List of all brick dependencies. In order to add a dependency, move the entry to 'dependencies'\\",
    \\"@coremedia/brick-360-spinner\\": \\"^1.0.0\\",
    \\"@coremedia/brick-bootstrap\\": \\"^1.0.0\\",
    \\"@coremedia/brick-cta\\": \\"^1.0.0\\",
    \\"@coremedia/brick-default-teaser\\": \\"^1.0.0\\",
    \\"@coremedia/brick-download-portal\\": \\"^1.0.0\\",
    \\"@coremedia/brick-dynamic-include\\": \\"^1.0.0\\",
    \\"@coremedia/brick-elastic-social\\": \\"^1.0.0\\",
    \\"@coremedia/brick-footer\\": \\"^1.0.0\\",
    \\"@coremedia/brick-fragment-scenario\\": \\"^1.0.0\\",
    \\"@coremedia/brick-generic-templates\\": \\"^1.0.0\\",
    \\"@coremedia/brick-hero-teaser\\": \\"^1.0.0\\",
    \\"@coremedia/brick-image-maps\\": \\"^1.0.0\\",
    \\"@coremedia/brick-legacy-templates\\": \\"^1.0.0\\",
    \\"@coremedia/brick-livecontext\\": \\"^1.0.0\\",
    \\"@coremedia/brick-navigation\\": \\"^1.0.0\\",
    \\"@coremedia/brick-pdp-augmentation\\": \\"^1.0.0\\",
    \\"@coremedia/brick-preview\\": \\"^1.0.0\\",
    \\"@coremedia/brick-quick-info\\": \\"^1.0.0\\",
    \\"@coremedia/brick-responsive-images\\": \\"^1.0.0\\",
    \\"@coremedia/brick-search\\": \\"^1.0.0\\",
    \\"@coremedia/brick-shoppable-video\\": \\"^1.0.0\\",
    \\"@coremedia/brick-video\\": \\"^1.0.0\\",
    \\"@coremedia/js-logger\\": \\"^1.0.0\\",
    \\"@coremedia/sass-utils\\": \\"^1.0.0\\",
    \\"jquery\\": \\"3.2.1\\"
  },
  \\"dependencies\\": {
    \\"@coremedia/cm-cli\\": \\"^2.0.0\\",
    \\"@coremedia/theme-utils\\": \\"^3.0.0\\",
    \\"webpack\\": \\"3.10.0\\"
  },
  \\"main\\": \\"src/js/index.js\\",
  \\"coremedia\\": {
    \\"type\\": \\"theme\\",
    \\"name\\": \\"newest\\"
  },
  \\"browserslist\\": [
    \\"last 1 Chrome version\\",
    \\"last 1 Firefox version\\",
    \\"last 1 Edge version\\",
    \\"Explorer >= 11\\"
  ]
}",
    "webpack.config.js": "const webpackTheme = require(\\"@coremedia/theme-utils/webpack.config.js\\");
module.exports = webpackTheme;
",
  },
  "/path/to/themes/newest-theme/src/js": Object {
    "index.js": "/*! Theme newest */
// activate brick js functionality
//import \\"@coremedia/brick-360-spinner\\";
//import \\"@coremedia/brick-bootstrap\\";
//import \\"@coremedia/brick-cta\\";
//import \\"@coremedia/brick-default-teaser\\";
//import \\"@coremedia/brick-download-portal\\";
//import \\"@coremedia/brick-dynamic-include\\";
//import \\"@coremedia/brick-elastic-social\\";
//import \\"@coremedia/brick-footer\\";
//import \\"@coremedia/brick-fragment-scenario\\";
//import \\"@coremedia/brick-generic-templates\\";
//import \\"@coremedia/brick-hero-teaser\\";
//import \\"@coremedia/brick-image-maps\\";
//import \\"@coremedia/brick-livecontext\\";
//import \\"@coremedia/brick-navigation\\";
//import \\"@coremedia/brick-pdp-augmentation\\";
//import \\"@coremedia/brick-preview\\";
//import \\"@coremedia/brick-quick-info\\";
//import \\"@coremedia/brick-responsive-images\\";
//import \\"@coremedia/brick-search\\";
//import \\"@coremedia/brick-shoppable-video\\";
//import \\"@coremedia/brick-video\\";
import \\"./newest.js\\";
",
    "newest.js": "//import $ from \\"jquery\\";
//import logger from \\"@coremedia/js-logger\\";
// --- JQUERY DOCUMENT READY -------------------------------------------------------------------------------------------
//$(function () {
//  logger.log(\\"Theme newest is used.\\");
//});
",
  },
  "/path/to/themes/newest-theme/src/l10n": Object {
    "Newest_en.properties": "",
  },
  "/path/to/themes/newest-theme/src/sass": Object {
    "newest.scss": "/*! Theme newest */
// ### VARIABLES ###

// Own variables (need to be loaded first, so default values can be overridden)
// @see http://sass-lang.com/documentation/file.SASS_REFERENCE.html#Variable_Defaults___default

//@import \\"variables/...\\";

// Dependency variables

//@import \\"~@coremedia/brick-bootstrap/src/sass/variables\\";
//@import \\"~@coremedia/brick-cta/src/sass/variables\\";
//@import \\"~@coremedia/brick-default-teaser/src/sass/variables\\";
//@import \\"~@coremedia/brick-download-portal/src/sass/variables\\";
//@import \\"~@coremedia/brick-dynamic-include/src/sass/variables\\";
//@import \\"~@coremedia/brick-elastic-social/src/sass/variables\\";
//@import \\"~@coremedia/brick-fragment-scenario/src/sass/variables\\";
//@import \\"~@coremedia/brick-generic-templates/src/sass/variables\\";
//@import \\"~@coremedia/brick-hero-teaser/src/sass/variables\\";
//@import \\"~@coremedia/brick-image-maps/src/sass/variables\\";
//@import \\"~@coremedia/brick-livecontext/src/sass/variables\\";
//@import \\"~@coremedia/brick-pdp-augmentation/src/sass/variables\\";
//@import \\"~@coremedia/brick-preview/src/sass/variables\\";
//@import \\"~@coremedia/brick-quick-info/src/sass/variables\\";
//@import \\"~@coremedia/brick-responsive-images/src/sass/variables\\";
//@import \\"~@coremedia/brick-shoppable-video/src/sass/variables\\";
//@import \\"~@coremedia/brick-video/src/sass/variables\\";
//@import \\"~@coremedia/sass-utils/src/variables\\";

// ### PARTIALS ###

// Dependency partials

//@import \\"~@coremedia/brick-bootstrap/src/sass/partials\\";
//@import \\"~@coremedia/brick-cta/src/sass/partials\\";
//@import \\"~@coremedia/brick-default-teaser/src/sass/partials\\";
//@import \\"~@coremedia/brick-download-portal/src/sass/partials\\";
//@import \\"~@coremedia/brick-dynamic-include/src/sass/partials\\";
//@import \\"~@coremedia/brick-elastic-social/src/sass/partials\\";
//@import \\"~@coremedia/brick-fragment-scenario/src/sass/partials\\";
//@import \\"~@coremedia/brick-generic-templates/src/sass/partials\\";
//@import \\"~@coremedia/brick-hero-teaser/src/sass/partials\\";
//@import \\"~@coremedia/brick-image-maps/src/sass/partials\\";
//@import \\"~@coremedia/brick-livecontext/src/sass/partials\\";
//@import \\"~@coremedia/brick-pdp-augmentation/src/sass/partials\\";
//@import \\"~@coremedia/brick-preview/src/sass/partials\\";
//@import \\"~@coremedia/brick-quick-info/src/sass/partials\\";
//@import \\"~@coremedia/brick-responsive-images/src/sass/partials\\";
//@import \\"~@coremedia/brick-shoppable-video/src/sass/partials\\";
//@import \\"~@coremedia/brick-video/src/sass/partials\\";
//@import \\"~@coremedia/sass-utils/src/partials\\";

// Own partials

//@import \\"partials/...\\";
",
    "preview.scss": "/*! Theme newest: Preview Styles */
// ### VARIABLES ###
// Own variables (need to be loaded first, so default values can be overridden)
// @see http://sass-lang.com/documentation/file.SASS_REFERENCE.html#Variable_Defaults___default
//@import \\"variables/...\\";
// Dependency variables
//@import \\"~@coremedia/brick-preview/src/sass/variables\\";
//@import \\"~@coremedia/sass-utils/src/variables\\";
// ### PARTIALS ###
// Dependency partials
//@import \\"~@coremedia/sass-utils/src/partials\\";
//@import \\"~@coremedia/brick-preview/src/sass/partials\\";
// Own partials
//@import \\"partials/preview\\";
",
  },
}
`;

exports[`createFolderStructure() creates folder structure of a new theme 1`] = `
Array [
  "/path/to/themes/newest-theme/src/js",
  "/path/to/themes/newest-theme/src/sass",
  "/path/to/themes/newest-theme/src/img",
  "/path/to/themes/newest-theme/src/fonts",
  "/path/to/themes/newest-theme/src/l10n",
  "/path/to/themes/newest-theme/src/templates/com.coremedia.blueprint.common.contentbeans",
]
`;

exports[`createTheme() creates folder structure and files of a new theme 1`] = `
Array [
  "/path/to/themes/newest-theme/src/js",
  "/path/to/themes/newest-theme/src/sass",
  "/path/to/themes/newest-theme/src/img",
  "/path/to/themes/newest-theme/src/fonts",
  "/path/to/themes/newest-theme/src/l10n",
  "/path/to/themes/newest-theme/src/templates/com.coremedia.blueprint.common.contentbeans",
]
`;

exports[`createTheme() creates folder structure and files of a new theme 2`] = `
Object {
  "/path/to/themes/newest-theme": Object {
    "newest-theme.xml": "<?xml version=\\"1.0\\" encoding=\\"UTF-8\\"?>
<themeDefinition modelVersion=\\"1\\">
<name>newest</name>
<!-- <description>a short description</description> -->
<!-- <thumbnail>img/theme-preview.jpg</thumbnail> -->
<javaScriptLibraries>
  <!-- add thirdparty js files that are not bundled via webpack here -->
  <!-- <javaScript>vendor/some-vendor-script.js</javaScript> -->
</javaScriptLibraries>
<javaScripts>
  <!-- add own js files that are not bundled via webpack here -->
  <!-- <javaScript>js/your-code.js</javaScript> -->
  <javaScript disableCompress=\\"true\\">js/newest.js</javaScript>
</javaScripts>
<styleSheets>
  <css disableCompress=\\"true\\">css/newest.css</css>
</styleSheets>
<templateSets>
  <templateSet>templates/newest-templates.jar</templateSet>
  <templateSet>templates/bricks-templates.jar</templateSet>
</templateSets>
<resourceBundles>
  <!-- add theme resource bundles here -->
  <resourceBundle>l10n/Newest_en.properties</resourceBundle>
  <!-- merged resource bundles of all bricks, activate this if you are using bricks -->
  <!-- <resourceBundle>l10n/Bricks_en.properties</resourceBundle> -->
</resourceBundles>
</themeDefinition>
",
    "package.json": "{
  \\"name\\": \\"@coremedia/newest-theme\\",
  \\"version\\": \\"1.0.0\\",
  \\"private\\": true,
  \\"scripts\\": {
    \\"build\\": \\"webpack\\",
    \\"start\\": \\"cm monitor\\",
    \\"theme-importer\\": \\"cm theme-importer\\"
  },
  \\"__comment__dependencies__\\": {
    \\"__comment__\\": \\"List of all brick dependencies. In order to add a dependency, move the entry to 'dependencies'\\",
    \\"@coremedia/brick-360-spinner\\": \\"^1.0.0\\",
    \\"@coremedia/brick-bootstrap\\": \\"^1.0.0\\",
    \\"@coremedia/brick-cta\\": \\"^1.0.0\\",
    \\"@coremedia/brick-default-teaser\\": \\"^1.0.0\\",
    \\"@coremedia/brick-download-portal\\": \\"^1.0.0\\",
    \\"@coremedia/brick-dynamic-include\\": \\"^1.0.0\\",
    \\"@coremedia/brick-elastic-social\\": \\"^1.0.0\\",
    \\"@coremedia/brick-footer\\": \\"^1.0.0\\",
    \\"@coremedia/brick-fragment-scenario\\": \\"^1.0.0\\",
    \\"@coremedia/brick-generic-templates\\": \\"^1.0.0\\",
    \\"@coremedia/brick-hero-teaser\\": \\"^1.0.0\\",
    \\"@coremedia/brick-image-maps\\": \\"^1.0.0\\",
    \\"@coremedia/brick-legacy-templates\\": \\"^1.0.0\\",
    \\"@coremedia/brick-livecontext\\": \\"^1.0.0\\",
    \\"@coremedia/brick-navigation\\": \\"^1.0.0\\",
    \\"@coremedia/brick-pdp-augmentation\\": \\"^1.0.0\\",
    \\"@coremedia/brick-preview\\": \\"^1.0.0\\",
    \\"@coremedia/brick-quick-info\\": \\"^1.0.0\\",
    \\"@coremedia/brick-responsive-images\\": \\"^1.0.0\\",
    \\"@coremedia/brick-search\\": \\"^1.0.0\\",
    \\"@coremedia/brick-shoppable-video\\": \\"^1.0.0\\",
    \\"@coremedia/brick-video\\": \\"^1.0.0\\",
    \\"@coremedia/js-logger\\": \\"^1.0.0\\",
    \\"@coremedia/sass-utils\\": \\"^1.0.0\\",
    \\"jquery\\": \\"3.2.1\\"
  },
  \\"dependencies\\": {
    \\"@coremedia/cm-cli\\": \\"^2.0.0\\",
    \\"@coremedia/theme-utils\\": \\"^3.0.0\\",
    \\"webpack\\": \\"3.10.0\\"
  },
  \\"main\\": \\"src/js/index.js\\",
  \\"coremedia\\": {
    \\"type\\": \\"theme\\",
    \\"name\\": \\"newest\\"
  },
  \\"browserslist\\": [
    \\"last 1 Chrome version\\",
    \\"last 1 Firefox version\\",
    \\"last 1 Edge version\\",
    \\"Explorer >= 11\\"
  ]
}",
    "webpack.config.js": "const webpackTheme = require(\\"@coremedia/theme-utils/webpack.config.js\\");
module.exports = webpackTheme;
",
  },
  "/path/to/themes/newest-theme/src/js": Object {
    "index.js": "/*! Theme newest */
// activate brick js functionality
//import \\"@coremedia/brick-360-spinner\\";
//import \\"@coremedia/brick-bootstrap\\";
//import \\"@coremedia/brick-cta\\";
//import \\"@coremedia/brick-default-teaser\\";
//import \\"@coremedia/brick-download-portal\\";
//import \\"@coremedia/brick-dynamic-include\\";
//import \\"@coremedia/brick-elastic-social\\";
//import \\"@coremedia/brick-footer\\";
//import \\"@coremedia/brick-fragment-scenario\\";
//import \\"@coremedia/brick-generic-templates\\";
//import \\"@coremedia/brick-hero-teaser\\";
//import \\"@coremedia/brick-image-maps\\";
//import \\"@coremedia/brick-livecontext\\";
//import \\"@coremedia/brick-navigation\\";
//import \\"@coremedia/brick-pdp-augmentation\\";
//import \\"@coremedia/brick-preview\\";
//import \\"@coremedia/brick-quick-info\\";
//import \\"@coremedia/brick-responsive-images\\";
//import \\"@coremedia/brick-search\\";
//import \\"@coremedia/brick-shoppable-video\\";
//import \\"@coremedia/brick-video\\";
import \\"./newest.js\\";
",
    "newest.js": "//import $ from \\"jquery\\";
//import logger from \\"@coremedia/js-logger\\";
// --- JQUERY DOCUMENT READY -------------------------------------------------------------------------------------------
//$(function () {
//  logger.log(\\"Theme newest is used.\\");
//});
",
  },
  "/path/to/themes/newest-theme/src/l10n": Object {
    "Newest_en.properties": "",
  },
  "/path/to/themes/newest-theme/src/sass": Object {
    "newest.scss": "/*! Theme newest */
// ### VARIABLES ###

// Own variables (need to be loaded first, so default values can be overridden)
// @see http://sass-lang.com/documentation/file.SASS_REFERENCE.html#Variable_Defaults___default

//@import \\"variables/...\\";

// Dependency variables

//@import \\"~@coremedia/brick-bootstrap/src/sass/variables\\";
//@import \\"~@coremedia/brick-cta/src/sass/variables\\";
//@import \\"~@coremedia/brick-default-teaser/src/sass/variables\\";
//@import \\"~@coremedia/brick-download-portal/src/sass/variables\\";
//@import \\"~@coremedia/brick-dynamic-include/src/sass/variables\\";
//@import \\"~@coremedia/brick-elastic-social/src/sass/variables\\";
//@import \\"~@coremedia/brick-fragment-scenario/src/sass/variables\\";
//@import \\"~@coremedia/brick-generic-templates/src/sass/variables\\";
//@import \\"~@coremedia/brick-hero-teaser/src/sass/variables\\";
//@import \\"~@coremedia/brick-image-maps/src/sass/variables\\";
//@import \\"~@coremedia/brick-livecontext/src/sass/variables\\";
//@import \\"~@coremedia/brick-pdp-augmentation/src/sass/variables\\";
//@import \\"~@coremedia/brick-preview/src/sass/variables\\";
//@import \\"~@coremedia/brick-quick-info/src/sass/variables\\";
//@import \\"~@coremedia/brick-responsive-images/src/sass/variables\\";
//@import \\"~@coremedia/brick-shoppable-video/src/sass/variables\\";
//@import \\"~@coremedia/brick-video/src/sass/variables\\";
//@import \\"~@coremedia/sass-utils/src/variables\\";

// ### PARTIALS ###

// Dependency partials

//@import \\"~@coremedia/brick-bootstrap/src/sass/partials\\";
//@import \\"~@coremedia/brick-cta/src/sass/partials\\";
//@import \\"~@coremedia/brick-default-teaser/src/sass/partials\\";
//@import \\"~@coremedia/brick-download-portal/src/sass/partials\\";
//@import \\"~@coremedia/brick-dynamic-include/src/sass/partials\\";
//@import \\"~@coremedia/brick-elastic-social/src/sass/partials\\";
//@import \\"~@coremedia/brick-fragment-scenario/src/sass/partials\\";
//@import \\"~@coremedia/brick-generic-templates/src/sass/partials\\";
//@import \\"~@coremedia/brick-hero-teaser/src/sass/partials\\";
//@import \\"~@coremedia/brick-image-maps/src/sass/partials\\";
//@import \\"~@coremedia/brick-livecontext/src/sass/partials\\";
//@import \\"~@coremedia/brick-pdp-augmentation/src/sass/partials\\";
//@import \\"~@coremedia/brick-preview/src/sass/partials\\";
//@import \\"~@coremedia/brick-quick-info/src/sass/partials\\";
//@import \\"~@coremedia/brick-responsive-images/src/sass/partials\\";
//@import \\"~@coremedia/brick-shoppable-video/src/sass/partials\\";
//@import \\"~@coremedia/brick-video/src/sass/partials\\";
//@import \\"~@coremedia/sass-utils/src/partials\\";

// Own partials

//@import \\"partials/...\\";
",
    "preview.scss": "/*! Theme newest: Preview Styles */
// ### VARIABLES ###
// Own variables (need to be loaded first, so default values can be overridden)
// @see http://sass-lang.com/documentation/file.SASS_REFERENCE.html#Variable_Defaults___default
//@import \\"variables/...\\";
// Dependency variables
//@import \\"~@coremedia/brick-preview/src/sass/variables\\";
//@import \\"~@coremedia/sass-utils/src/variables\\";
// ### PARTIALS ###
// Dependency partials
//@import \\"~@coremedia/sass-utils/src/partials\\";
//@import \\"~@coremedia/brick-preview/src/sass/partials\\";
// Own partials
//@import \\"partials/preview\\";
",
  },
}
`;
